airflow/providers/amazon/aws/operators/sagemaker.py

```diff
@@ -0,0 +1,643 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import json
+import sys
+from typing import List, Optional
+
+from botocore.exceptions import ClientError
+
+from airflow.exceptions import AirflowException
+from airflow.models import BaseOperator
+from airflow.providers.amazon.aws.hooks.base_aws import AwsBaseHook
+from airflow.providers.amazon.aws.hooks.sagemaker import SageMakerHook
+
+if sys.version_info >= (3, 8):
+    from functools import cached_property
+else:
+    from cached_property import cached_property
```

Comment:
I have seen this also done as a try/except block elsewhere in the Airflow codebase.  Do we have a preference on one form over another?

Reply:
We are not changing the code of the operators in this PR. This is just about classes names/files. 
If the operator code needs some work it should be done seperatly.

Specifically here - in airflow code base we do this import from compact but in providers we cant because we dont want to limit the provider only to new airflow versions. Eventually this import will be replaced with:
`from airflow.compat.functools import cached_property`

For the moment we are OK here :)